src/main/kotlin/com/hibob/academy/employeeFeedback/resource/FeedbackResource.kt

```diff
@@ -0,0 +1,88 @@
+package com.hibob.academy.employeeFeedback.resource
+
+import com.hibob.academy.employeeFeedback.dao.AnonymityType.Companion.convertStringToAnonymityType
+import com.hibob.academy.employeeFeedback.dao.FeedbackSubmitRequest
+import com.hibob.academy.employeeFeedback.dao.RoleType
+import com.hibob.academy.employeeFeedback.dao.RoleType.Companion.convertStringToRoleType
+import jakarta.ws.rs.core.Response
+import com.hibob.academy.employeeFeedback.service.FeedbackService
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.ws.rs.*
+import jakarta.ws.rs.container.ContainerRequestContext
+import jakarta.ws.rs.core.Context
+import jakarta.ws.rs.core.MediaType
+import org.springframework.stereotype.Controller
+import org.springframework.web.bind.annotation.RequestBody
+
+
+@Controller
+@Path("/api/feedback")
+@Produces(MediaType.APPLICATION_JSON)
+class FeedbackResource(private val feedbackService: FeedbackService) {
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    fun submitFeedback(@RequestBody feedbackRequest: FeedbackSubmitRequest, @Context request: ContainerRequestContext): Response {
+        val employeeId = getLoggedInEmployeeId(request)
+        val companyId = getLoggedInCompanyId(request)
+        val anonymity = convertStringToAnonymityType(feedbackRequest.anonymity)
+
+        val success = feedbackService.submitFeedback(employeeId, companyId, anonymity, feedbackRequest.feedback)
+
+        if (success) return Response.ok("Feedback created Successfully").build()
+
+        return Response.status(Response.Status.BAD_REQUEST).entity("Unable to create feedback").build()
+    }
+
+    @GET
+    @Path("/allFeedback")
+    fun viewFeedback(@Context request: ContainerRequestContext): Response {
+        val employeeId = getLoggedInEmployeeId(request)
+        val companyId = getLoggedInCompanyId(request)
+        val role = getLoggedInRole(request)
+
+        if (!validateRole(role)) return Response.status(Response.Status.UNAUTHORIZED).build()
+
+        val allFeedback = feedbackService.viewAllSubmittedFeedback(employeeId, companyId)
+
+        return Response.ok(allFeedback).build()
+    }
+
+    @GET
+    @Path("/checkStatus")
+    fun checkFeedbackStatus(@QueryParam ("feedbackId") feedbackId: Int?, @Context request: ContainerRequestContext): Response {
+        val employeeId = getLoggedInEmployeeId(request)
+        val companyId = getLoggedInCompanyId(request)
+
+        feedbackId?.let {
+
+            return Response.ok(feedbackService.viewStatusOfMyFeedback(employeeId, companyId, feedbackId)).build()
```

Comment:
instead of doing this if, maybe you can change a bit the service, so it can handle recieving null as a param

Reply:
changed it :)